### DISCUSSION TITLE
Multiple Uyuni Proxy Servers Registered To One Master

### DISCUSSION THREAD

**Discussion by nelsinchi** [uyuni-project/uyuni#5173] [Q&A]
Hi There,

Is it possible to add multiple Uyuni Proxy Servers registered to only one Uyuni Master Server? If yes, what would be the procedure to follow?

Thanks a lot in advance!

**Comment by mcalmer** [comment]
Sure, you can. 
Check the documentation about installation and setup
https://www.uyuni-project.org/uyuni-docs/en/uyuni/installation-and-upgrade/install-proxy-uyuni.html

**Comment by nelsinchi** [comment]
Hi @mcalmer , I have asked that, because I have already helped to implement 2 Environments and 1 extra in my own, and for the three cases we haven't been able to setup a secondary Uyuni Proxy Server even if we use closely the Installation and Setup Documentation from the same link that you are providing.

For all the three cases we complete the Proxy Installation and Setup procedure, but when we accept the key through the Web Portal, the key is accepted, but the server name doesn´t become a hyperlink like the others accepted keys/servers, and cannot be findable on the Server List.

![vivaldi_mXbfe5H5kN](https://user-images.githubusercontent.com/88047647/162471609-09b220e0-759b-4234-a00a-90dd438151b8.png)


Thanks a lot in advance!
Kind regards.

**Comment by mcalmer** [comment]
Maybe you use cloned imaged? In that case 1 maybe replace the other.
Please read:

https://www.uyuni-project.org/uyuni-docs/en/uyuni/administration/tshoot-registerclones.html

**Reply by nelsinchi** [reply]
No, each Proxy Server was built separately.

**Reply by nelsinchi** [reply]
By the way, I am going to follow your troubleshoot's process and cross my fingers. I will let you know my result soon.

Thanks.

**Reply by nelsinchi** [reply]
I followed the procedure and still without luck! :(

After that, I checked the salt-minion status, and this below is what I get:


![WindowsTerminal_l5pKDwyiOH](https://user-images.githubusercontent.com/88047647/162482049-c19645e9-9bd3-442a-845e-a562d01184b9.png)

Thanks.

**Reply by nelsinchi** [reply]
The new Proxy Server is not getting registered after the key is accepted. I receive the following lines in the log:

```
Apr 08 11:35:10 uyunipxdmz-server salt-minion[14871]:     raise_exc_info(self._exc_info)
Apr 08 11:35:10 uyunipxdmz-server salt-minion[14871]:   File "<string>", line 4, in raise_exc_info
Apr 08 11:35:10 uyunipxdmz-server salt-minion[14871]:   File "/usr/lib/python3.6/site-packages/salt/ext/tornado/gen.py", line 1070, in run
Apr 08 11:35:10 uyunipxdmz-server salt-minion[14871]:     yielded = self.gen.send(value)
Apr 08 11:35:10 uyunipxdmz-server salt-minion[14871]:   File "/usr/lib/python3.6/site-packages/salt/crypt.py", line 777, in sign_in
Apr 08 11:35:10 uyunipxdmz-server salt-minion[14871]:     ret = self.handle_signin_response(sign_in_payload, payload)
Apr 08 11:35:10 uyunipxdmz-server salt-minion[14871]:   File "/usr/lib/python3.6/site-packages/salt/crypt.py", line 789, in handle_signin_response
Apr 08 11:35:10 uyunipxdmz-server salt-minion[14871]:     clear_signature = payload["sig"]
Apr 08 11:35:10 uyunipxdmz-server salt-minion[14871]: KeyError: 'sig'
Apr 08 11:36:10 uyunipxdmz-server salt-minion[14871]: [ERROR   ] Minion unable to successfully connect to a Salt Master.
```

Thanks.

**Comment by mcalmer** [comment]
Sounds like the last security fix for salt. Please check if the server has the latest salt-master installed. The master must be newer or at least the same version as the minion

**Reply by nelsinchi** [reply]
Okay. I am going to upgrade the Uyuni Master Server by following the documentation this afternoon and then let's see how it goes.

Thanks a lot!

**Reply by nelsinchi** [reply]
Hi @mcalmer, I do have some good news!

When I performed the Major Upgrade on the Uyuni Master Server. After that, I registered once again the new Uyuni Proxy Server from scratch with the Uyuni Master Server.

![vivaldi_cLiGSr58zi](https://user-images.githubusercontent.com/88047647/162550865-d7da1658-4721-471d-a13d-cdf17870fb11.png)

After that, the server started appearing in the System List as expected, then I subscribed it with the channels from openSUSE, then I finished the New Proxy Server Setup by following the documentation.

Finally, the new Proxy is as it should be.

![mstsc_nw03cBr59u](https://user-images.githubusercontent.com/88047647/162550990-2b9cea18-9638-47ee-b978-9564b95905be.png)

![mstsc_qcF0HhfHLG](https://user-images.githubusercontent.com/88047647/162551022-3ed448ba-07d9-4e57-b055-254bd8370564.png)

In the end, I can confirm that the root cause of the error was related to the missing Major Upgrade on the Uyuni Master Server.

Thanks a lot @mcalmer for all your support.
